modules/javafx.graphics/src/main/java/javafx/application/Platform.java

```diff
@@ -432,4 +436,280 @@ public static ReadOnlyBooleanProperty accessibilityActiveProperty() {
         }
         return accessibilityActiveProperty.getReadOnlyProperty();
     }
+
+    /**
+     * Gets the preferences of the current platform.
+     * <p>
+     * The map returned from this method is unmodifiable, which means that keys and values cannot
+     * be added, removed, or updated. Calling any mutator method on the map will always cause
+     * {@code UnsupportedOperationException} to be thrown. However, the mappings will be updated
+     * by JavaFX when the operating system reports that a platform preference has changed.
+     *
+     * @return the {@code Preferences} instance
+     * @see <a href="Platform.Preferences.html#preferences-table">List of platform preferences</a>
+     * @since 22
+     */
+    public static Preferences getPreferences() {
+        return PlatformImpl.getPlatformPreferences();
+    }
+
+    /**
+     * Contains UI preferences of the current platform.
+     * <p>
+     * {@code Preferences} extends {@link ObservableMap} to expose platform preferences as key-value pairs.
+     * For convenience, {@link #getInteger}, {@link #getDouble}, {@link #getBoolean}, {@link #getString},
+     * {@link #getColor}, {@link #getPaint}, and {@link #getValue} are provided as typed alternatives to
+     * the untyped {@link #get} method.
+     * <p>
+     * The preferences that are reported by the platform may be dependent on the operating system version,
+     * so applications should not assume that a particular preference is always available.
+     * <p>
+     * The following list contains all preferences that are potentially available on the specified platforms:
+     * <table id="preferences-table">
+     *     <caption>List of platform preferences</caption>
+     *     <tbody>
+     *         <tr><th colspan="2" scope="colgroup">Windows</th></tr>
+     *         <tr><td>{@code Windows.SPI.HighContrast}</td><td>{@link Boolean}</td></tr>
```

Comment:
would it be possible to sort the keys alphabetically?

Reply:
I don't think that's necessarily a good idea. I've reordered some of them, but others should be grouped. For example: it makes sense to keep the following keys together:
* `macOS.NSColor.labelColor`
* `macOS.NSColor.secondaryLabelColor`
* `macOS.NSColor.tertiaryLabelColor`
* `macOS.NSColor.quaternaryLabelColor`

Another example:
* `macOS.NSColor.textBackgroundColor`
* `macOS.NSColor.selectedTextBackgroundColor`

As it is, most of the keys appear in the same order as they appear in the respective platform's developer documentation.

Reply:
makes sense, thank you for clarification.